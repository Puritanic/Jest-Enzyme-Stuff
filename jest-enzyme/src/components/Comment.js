import React from 'react';

const Comment = ({ comment }) => {
	return (
		<ul className="comment">
			<p className="comment__content">{comment.content}</p>
			<small className="comment__author">{comment.author}</small>
		</ul>
	);
};

export default Comment;
