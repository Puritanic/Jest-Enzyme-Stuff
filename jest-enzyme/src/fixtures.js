export const comments = [
	{ id: 67829, content: 'Just a comment', author: 'PuritanicC' },
	{ id: 70501, content: 'Just a comment', author: 'PuritanicA' },
	{ id: 19898, content: 'Just a comment', author: 'PuritanicB' },
];
