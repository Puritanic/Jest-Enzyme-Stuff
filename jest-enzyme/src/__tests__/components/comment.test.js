/* eslint-disable no-undef */
import React from 'react';

import { comments } from '../../fixtures';
import Comment from '../../components/Comment';

describe('<Comment />', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallow(<Comment comment={comments[0]} />);
	});

	it('should match snapshot', () => {
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('should render correct data', () => {
		expect(wrapper.find('.comment__content').text()).toEqual(comments[0].content);
		expect(wrapper.find('.comment__author').text()).toEqual(comments[0].author);
	});
});
