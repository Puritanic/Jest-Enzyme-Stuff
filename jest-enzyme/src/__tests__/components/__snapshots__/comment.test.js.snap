// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comment /> should match snapshot 1`] = `
<ul
  className="comment"
>
  <p
    className="comment__content"
  >
    Just a comment
  </p>
  <small
    className="comment__author"
  >
    PuritanicC
  </small>
</ul>
`;
